### PR TITLE
Add jdk 11 support

### LINKED ISSUE #1998: Add JDK11+ compatibility
**Describe the issue**
Some classes are not compatible with newer JDK versions.
Create a wrapper that tries the old, then new version then throws exception if it cannot find the class.

**Reporter**
Core-Team

### PR DESCRIPTION
**Pull request by alisihab**
Closes #1998

### CODE REVIEW

File: test/pom.xml [hunk 1]
```diff
@@ -35,7 +35,7 @@
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-compiler-plugin</artifactId>
-				<version>3.1</version>
+				<version>3.8.1</version>
```
Comment (nielsm5) [inline test/pom.xml:38]: Can this be managed via pluginManagement?

File: test/pom.xml [hunk 2]
```diff
@@ -115,7 +115,7 @@
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-war-plugin</artifactId>
-				<version>2.6</version>
+				<version>3.2.2</version>
```
Comment (nielsm5) [inline test/pom.xml:118]: Can this be managed via pluginManagement?

File: test/pom.xml [hunk 3]
```diff
@@ -194,5 +194,23 @@
...
+						<groupId>org.apache.maven.plugins</groupId>
+						<artifactId>maven-compiler-plugin</artifactId>
+						<version>3.8.1</version>
+						<configuration>
+							<release>11</release>
+						</configuration>
+					</plugin>
+				</plugins>
+			</build>
+		</profile>
```
Comment (nielsm5) [inline test/pom.xml:214]: I would expect this in the parent pom, and other poms to inherit it from there. Does that not work?

File: ibm/pom.xml [hunk 4]
```diff
@@ -1,65 +1,94 @@
...
+			<activation>
+				<jdk>11</jdk>
+			</activation>
+			<dependencies>
+				<dependency>
+					<groupId>com.ibm</groupId>
+					<artifactId>ws-runtime</artifactId>
+					<version>8.0.0</version>
+					<scope>provided</scope>
+				</dependency>
```
Comment (nielsm5) [inline ibm/pom.xml:78]: This doesn't have to be in a profile. You can upgrade the old was6 dependency

File: core/pom.xml [hunk 5]
```diff
@@ -680,5 +680,75 @@
...
+					<version>${cxf.version}</version>
+					<exclusions>
+						<exclusion>
+							<groupId>com.sun.xml.messaging.saaj</groupId>
+							<artifactId>saaj-impl</artifactId>
+						</exclusion>
+						<exclusion>
+							<groupId>org.apache.geronimo.specs</groupId>
+							<artifactId>geronimo-jta_1.1_spec</artifactId>
+						</exclusion>
```
Comment (nielsm5) [inline core/pom.xml:736]: Is it possible to exclude these by default? Do we need the JTA spec compile time?

**Review by nielsm5 — CHANGES_REQUESTED** [review]

**Review by nielsm5 — APPROVED** [review]

**Review by gvanbrakel — APPROVED** [review]

**Pull request merged**